File: src/main/java/com/redeprivada/sorting/Insertionsort.java
package com.redeprivada.sorting;

public class Insertionsort implements Sortable {
    @Override
    public void sort(int[] nums) {
        for (int i = 1; i < nums.length; i++) {
            int key = nums[i];
            int j = i - 1;
            while(j >= 0 && nums[j] > key) {
                nums[j+1] = nums[j];
                j = j - 1;
            }
            nums[j + 1] = key;
        }
    }
}

File: src/main/kotlin/com/redeprivada/sorting/InsertionSort.kt
package com.redeprivada.sorting

class InsertionSort : Sortable {
    override fun sort(nums: IntArray) {
        for (i in 1..< nums.size) {
            val key = nums[i]
            var j = i - 1
            while (j >= 0 && nums[j] > key) {
                nums[j + 1] = nums[j]
                j -= 1
            }
            nums[j + 1] = key
        }
    }
}